staging/src/k8s.io/apimachinery/pkg/apis/meta/v1/meta.go

```diff
@@ -168,3 +170,14 @@ func (meta *ObjectMeta) SetOwnerReferences(references []OwnerReference) {
 }
 func (meta *ObjectMeta) GetClusterName() string            { return meta.ClusterName }
 func (meta *ObjectMeta) SetClusterName(clusterName string) { meta.ClusterName = clusterName }
+
+func (meta *ObjectMeta) GetManagedFields() map[string]VersionedFields {
+	return meta.ManagedFields
+}
+
+func (meta *ObjectMeta) SetManagedFields(ManagedFields map[string]VersionedFields) {
+	meta.ManagedFields = make(map[string]VersionedFields, len(ManagedFields))
+	for key, value := range ManagedFields {
+		meta.ManagedFields[key] = value
```

Comment:
Is it OK that this doesn't do a deep copy? If a shallow copy is OK why is it necessary to copy at all?